kimchi/src/tests/framework.rs

```diff
@@ -134,4 +139,51 @@ impl TestFramework {
         verify::<Affine, BaseSponge, ScalarSponge>(&group_map, &verifier_index, &proof).unwrap();
         println!("- time to verify: {}ms", start.elapsed().as_millis());
     }
+
+    /// Create and verify proof with deserialization
+    pub(crate) fn run_test_serialization(
+        gates: Vec<CircuitGate<Fp>>,
+        witness: [Vec<Fp>; COLUMNS],
+        public: &[Fp],
+    ) {
+        // create the index
+        let start = Instant::now();
+        let index = new_index_for_test(gates, public.len());
+        let verifier_index = index.verifier_index();
+        println!("- time to create index: {:?}s", start.elapsed().as_secs());
+
+        let verifier_index_serialize =
+            serde_json::to_string(&verifier_index).expect("couldn't serialize index");
```

Comment:
thinking out loud: wondering how big the `.json` will be :o will it even be readable? (I'm assuming a lot of the binary stuff doesn't encode nicely to json, and will take a huge amount of space)